src/server/data_export.py

```diff
@@ -0,0 +1,59 @@
+#
+# Data export handlers
+#
+
+#
+# Data exporters first collect data via their assembler function,
+# then pass that data through the formatter function before output.
+# PageCache is updated before export() is called and can be assumed accurate.
+#
+# name should be unique and acts as an identifier
+# label becomes visible in the UI and becomes translatable
+# formatter(data) should be used for final handling of file format
+# assembler(Database, PageCache) collects relevant data from timer
+#   before formatting.
+#
+
+import logging
+from Plugins import Plugins
+
+class dataExportManager():
+    exporters = {}
+
+    def __init__(self):
+        self.readPlugins()
+
+    def readPlugins(self):
+        exporter_plugins = Plugins(prefix='data_export')
+        exporter_plugins.discover()
+
+        for exporter in exporter_plugins:
+            self.registerExporter(exporter)
+
+    def registerExporter(self, exporter):
+        if hasattr(exporter, 'name'):
+            if exporter.name in self.exporters:
+                logger.warning('Overwriting data exporter "{0}"'.format(exporter['name']))
+
+            self.exporters[exporter.name] = exporter
+        else:
+            logger.warning('Invalid exporter')
+
+    def isExporter(self, exporter_id):
```

Comment:
hasExporter is probably a better name. I think exporter_id should be exporter_name to be consistent with elsewhere.